db/virtual_tables.hh

```diff
@@ -38,4 +39,16 @@ future<> initialize_virtual_tables(
     sharded<db::system_keyspace>& sys_ks,
     db::config&);
 
+
+class virtual_table;
+
+using virtual_tables_registry_impl = std::map<table_id, std::unique_ptr<virtual_table>>;
```

Comment:
It's partially defeats the purpose of pimpl that we need to include <map> here, although I'm hoping the compiler doesn't actually need to compile any map code to implement unique_ptr<map<...>>. 
Could virtual_tables_registry_impl be a class that *contains* a map, instead of being a map, which could allow us to forward-declare that class without including <map>?

Reply:
I don't think the compiler will instantiate the map, and certainly won't compile any of its methods, only when used.